### ISSUE TITLE
"eslint.onIgnoredFiles" seems not working on v3.0.10

### ISSUE THREAD

**Issue by zheung** [microsoft/vscode-eslint#1868]
After upgrading to v3.x, I found that `onIgnoredFiles` does not generate warnings properly:

settings.json
````json
{
	"eslint.onIgnoredFiles": "warn"
}
````
eslint.config.js
````javascript
export default [
	{ ignores: ['dist/**'] },
];
````
note:
- vscode-eslint version: `3.0.10` (also `3.0.8`)
- eslint version: `9.5.0`
- node version (local): `20.14.0`
- using flat config
- it correctly lints for files that have not been ignored

screeonshot on v3.0.10 (not working):
![20240617180545](https://github.com/microsoft/vscode-eslint/assets/894004/63217ac4-a1bf-4822-9c85-2a37e24df3cc)

screeonshot on v2.4.4 (working):
![20240617180410](https://github.com/microsoft/vscode-eslint/assets/894004/376a1928-dec9-4963-b486-82715f5ed603)
